components/datetime/src/pattern/error.rs

```diff
@@ -10,7 +10,7 @@ use displaydoc::Display;
 ///
 /// Serde will generate an error such as:
 /// "invalid value: unclosed literal in pattern, expected a valid UTS 35 pattern string at line 1 column 12"
-#[derive(Display, Debug, PartialEq, Copy, Clone)]
+#[derive(Display, Debug, PartialEq, Eq, Copy, Clone)]
```

Comment:
Why is everything getting `Eq`? It seems like an unnecessarily restrictive bound. I can imagine wanting to add floats to our error outputs, for example.

Reply:
It's a new clippy lint - https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq

I imagine if we were to add floats to a type, we'd remove `Eq`.

Reply:
Except that removing Eq is a breaking change

Reply:
I feel like the lint should not apply to non_exhaustive structs. Can we fix it that way?

Reply:
I can manually disable it on every struct that is non_exhaustive, but that would be daunting.

@Manishearth wdyt?

Reply:
The lint was downgraded to `nursery`, so it's off by default. We should global-allow it.